module/x/gravity/keeper/ethereum_event_handler.go

```diff
@@ -121,3 +78,102 @@ func (a EthereumEventProcessor) Handle(ctx sdk.Context, eve types.EthereumEvent)
 	}
 	return nil
 }
+
+func (a EthereumEventProcessor) verifyERC20DeployedEvent(ctx sdk.Context, event *types.ERC20DeployedEvent) error {
+	if existingERC20, exists := a.keeper.getCosmosOriginatedERC20(ctx, event.CosmosDenom); exists {
+		return sdkerrors.Wrapf(
+			types.ErrInvalid,
+			"ERC20 token %s already exists for denom %s", existingERC20.Hex(), event.CosmosDenom,
+		)
+	}
+
+	// We expect that all Cosmos-based tokens have metadata defined. In the case
+	// a token does not have metadata defined, e.g. an IBC token, we successfully
+	// handle the token under the following conditions:
+	//
+	// 1. The ERC20 name is equal to the token's denomination. Otherwise, this
+	// 		means that ERC20 tokens would have an untenable UX.
+	// 2. The ERC20 token has zero decimals as this is what we default to since
+	// 		we cannot know or infer the real decimal value for the Cosmos token.
+	// 3. The ERC20 symbol is empty.
```

Comment:
q: why not use the same token denom?

Reply:
I don't follow?

Reply:
nvm, I got confused with the other function